### DISCUSSION TITLE
Log stack trace for uncaught exceptions in request handlers

### DISCUSSION THREAD

**Discussion by jkieboom** [mswjs/msw#1416] [Ideas]
### Scope

Improves an existing behavior

### Compatibility

- [ ] This is a breaking change

### Feature description

It is nice that msw automatically catches exceptions in request handlers and responds with a 500. I found it a little hard to debug as the debug logging only includes the error name and message but not the error stack. https://github.com/mswjs/msw/blob/638fad9640885c0a85153c1b9ade61495d4295bb/src/setupWorker/start/createRequestListener.ts#L99-L109

It would be nice if this error logging also included the stack.

**Comment by kettanaito** [comment]
Hey, @jkieboom.

Have you inspected that 500 response in the network? It contains a JSON response body that includes a full stack of the caught error.

Do you wish for it to also be printed into the console?

**Comment by jkieboom** [comment]
Yes, the issue in my case is that my network requests go through a secondary library that transforms 500 responses into thrown errors but it doesn't know about special response payloads. So it would be helpful if it would also be printed.

**Comment by kettanaito** [comment]
I don't mind giving this a try. @jkieboom, would you have a chance to open a pull request?

**Comment by jkieboom** [comment]
https://github.com/mswjs/msw/pull/1417

**Reply by kettanaito** [reply]
Awesome! Thank you for working on this. I've left a few thoughts on the change, would like to know your opinion on those.
